libvcx/src/utils/devsetup_agent.rs

```diff
@@ -0,0 +1,462 @@
+            settings::clear_config();
+            let enterprise_seed = "000000000000000000000000Trustee1";
+            let config_wallet = WalletConfig {
+                wallet_name: format!("faber_wallet_{}", uuid::Uuid::new_v4().to_string()),
+                wallet_key: settings::DEFAULT_WALLET_KEY.into(),
+                wallet_key_derivation: settings::WALLET_KDF_RAW.into(),
+                wallet_type: None,
+                storage_config: None,
+                storage_credentials: None,
+                rekey: None,
+                rekey_derivation_method: None
+            };
+            let config_provision_agent = ProvisionAgentConfig {
+                agency_did: AGENCY_DID.to_string(),
+                agency_verkey: AGENCY_VERKEY.to_string(),
+                agency_endpoint: AGENCY_ENDPOINT.to_string(),
+                agent_seed: None
+            };
+            create_wallet(&config_wallet).unwrap();
+            open_as_main_wallet(&config_wallet).unwrap();
+            let config_issuer = configure_issuer_wallet(enterprise_seed).unwrap();
+            init_issuer_config(&config_issuer).unwrap();
+            let config_agency = provision_cloud_agent(&config_provision_agent).unwrap();
+            close_main_wallet().unwrap();
+            Faber {
+                is_active: false,
+                config_wallet,
+                config_agency,
+                config_issuer,
+                schema_handle: 0,
+                cred_def_handle: 0,
+                connection_handle: 0,
+                credential_handle: 0,
+                presentation_handle: 0,
+            }
+        }
+
+
+        pub fn create_schema(&mut self) {
+            self.activate().unwrap();
+            let did = String::from("V4SGRU86Z58d6TV7PBUe6f");
+            let data = r#"["name","date","degree", "empty_param"]"#.to_string();
+            let name: String = crate::utils::random::generate_random_schema_name();
+            let version: String = String::from("1.0");
+
+            self.schema_handle = schema::create_and_publish_schema("test_schema", did.clone(), name, version, data).unwrap();
+        }
+
+        pub fn create_credential_definition(&mut self) {
+            self.activate().unwrap();
+
+            let schema_id = schema::get_schema_id(self.schema_handle).unwrap();
+            let did = String::from("V4SGRU86Z58d6TV7PBUe6f");
+            let name = String::from("degree");
+            let tag = String::from("tag");
+
+            self.cred_def_handle = credential_def::create_and_publish_credentialdef(String::from("test_cred_def"), name, did.clone(), schema_id, tag, String::from("{}")).unwrap();
+        }
+
+        pub fn create_presentation_request(&self) -> u32 {
+            let requested_attrs = json!([
+                {"name": "name"},
+                {"name": "date"},
+                {"name": "degree"},
+                {"name": "empty_param", "restrictions": {"attr::empty_param::value": ""}}
```

Comment:
Why do we value-check only empty param? Seems like a rather special case

Reply:
Can't answer this right now, this looks like addition in PR but it was probably just moved. Changing this is out of scope of this pr

Reply:
Alright